### PR TITLE
Feat/create entry

### PR DESCRIPTION
**Pull request by me-jterhorst**
Add a multi step form
- with controlled inputs, 
- persistent entries to cycle through
- Add more tasks as you like
- Write a new reminder into local storage

### CODE REVIEW

File: client/src/pages/CreateAction.js [hunk 1]
```diff
@@ -1,25 +1,154 @@
...
+import HeaderActionGoForward from "../components/Card_components/HeaderActionGoForward";
 import SpeechInput from "../components/SpeechInput";
-import { useParams } from "react-router";
+import { useParams, useHistory } from "react-router";
+import { useState, useEffect } from "react";
 import FooterSubmit from "../components/Card_components/FooterSubmit";
 
 export default function CreateCase() {
+  /** VARIABLE SET UP */
+  const reminderArray = JSON.parse(localStorage.getItem("newEntry"));
```
Comment (mpagels) [inline client/src/pages/CreateAction.js:11]: You need to be careful here. What happens when there is no `newEntry` in the localStorage?
You could use the logical OR the check that.
```js
JSON.parse(localStorage.getItem("newEntry")) || [];
```
This reads like this: When the left side is not true, return the left side.

File: client/src/pages/CreateAction.js [hunk 2]
```diff
@@ -1,25 +1,154 @@
...
   const { number } = useParams();
+  const indexNumber = Number(number);
+  const history = useHistory();
+  const [verbInput, setVerbInput] = useState(
+    reminderArray.tasks[indexNumber - 1].verb
```
Comment (mpagels) [inline client/src/pages/CreateAction.js:16]: I thought `reminderArray` is an array. But in this line it looks like, it is an object? I'm confused. Can you explain?

File: client/src/pages/CreateAction.js [hunk 3]
```diff
@@ -1,25 +1,154 @@
...
+  );
+  const [actionInput, setActionInput] = useState(
+    reminderArray.tasks[indexNumber - 1].action
+  );
+  const [withInput, setWithInput] = useState(
+    reminderArray.tasks[indexNumber - 1].with
+  );
```
Comment (mpagels) [inline client/src/pages/CreateAction.js:23]: It is important to use 4 states for every data? Could you just use one state?
```js
const [inputs, setInputs] = useState({
    verb: reminderArray.tasks[indexNumber - 1].verb,
    action: reminderArray.tasks[indexNumber - 1].action,
    with: reminderArray.tasks[indexNumber - 1].with
    })
```

And then changing this accordingly.

File: client/src/pages/CreateAction.js [hunk 4]
```diff
@@ -1,25 +1,154 @@
...
+    reminderArray.tasks[indexNumber - 1] = taskObj;
+    reminderArray.tasks.push({
+      taskId: indexNumber + 1,
+      verb: "",
+      action: "",
+      with: "",
+    });
+
+    localStorage.removeItem("newEntry");
+    localStorage.setItem("newEntry", JSON.stringify(reminderArray));
```
Comment (namirsab) [inline client/src/pages/CreateAction.js:56]: I have the feeling here  you don't want add the reminderArray to `newEntry` but to `reminder`? 

Somehow in my local storage i only have this `reminder` key that is an object, and I never get an array:

image

**Review by namirsab — CHANGES_REQUESTED** [review]
@me-jterhorst lots of work here!
Please take a look to my comments

**Review by mpagels — CHANGES_REQUESTED** [review]
Hey Jakob, very good job. Please see my comments.

- Found, maybe a wrong installed package. (uuid-v4)
- Highly recommend using propTypes here instead of passing an empty string as the default value for a function prop.

Stopped the review, since Namir points a lot of things and don't wanted to create duplicates.

File: client/src/pages/CreateAction.js (hunk 1 above)
Comment (me-jterhorst) [reply client/src/pages/CreateAction.js:11]: The createCase will only be loaded once the createCase has run. CreateCase opens up the new Entry. That's why it shouldn't matter in my opinion

File: client/src/pages/CreateAction.js (hunk 2 above)
Comment (me-jterhorst) [reply client/src/pages/CreateAction.js:16]: My bad, I renamed it to a more sensible name

File: client/src/pages/CreateAction.js (hunk 1 above)
Comment (me-jterhorst) [reply client/src/pages/CreateAction.js:11]: The Create Action can only be accessed after a CreateCase that should logically prevent someone to create an action without a newEntry

Comment (mpagels) [reply client/src/pages/CreateAction.js:11]: Going here: https://neue-fische-capstone-in-cased-git-feat-createentry-me-jterhorst.vercel.app/create/1 without having anything typed before breaks the app

grafik

File: client/src/components/SpeechInput.js [hunk 5]
```diff
@@ -15,6 +21,8 @@ export default function SpeechInput({ identifier, label, isRequired = true }) {
           name={`${identifier}__${label}`}
           id={`${identifier}__${label}`}
           placeholder={label}
+          value={value}
+          onChange={onChange}
```
[the remark this answers is not in the document]
Comment (me-jterhorst) [reply client/src/components/SpeechInput.js:25]: I haven't worked on that feature yet. It will be resolved at a later stage

**Pull request merged**